src/merkle_tree.rs

```diff
@@ -1,6 +1,7 @@
 use sha3::{Digest, Sha3_256};
 
 type HashedData = [u8; 32];
+type Position = usize;
```

Comment:
You should generally use type aliases for long types, as its more valuable to know the real type than a generic name, making the dev look up what type it actually is.

Reply:
Good point edg. In retrospect that Type was a bit overkill. @dsocolobsky pointed the same thing out actually.